pkg/unpack/tag.go

```diff
@@ -0,0 +1,112 @@
+package unpack
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+const (
+	tagJSON   = "json"
+	tagUnpack = "unpack"
+	tagSep    = ","
+)
+
+var (
+	ErrTag      = errors.New(`unpack tag must have form "", "<value>", "skip", "<value>,skip"`)
+	ErrSkip     = errors.New("unpack skip tag can only appear once")
+	ErrNeedJSON = errors.New("unpack tag cannot appear without a JSON tag")
+)
+
+func parseTag(which string, f reflect.StructField) (string, bool, []string) {
+	tag, ok := f.Tag.Lookup(which)
+	if !ok {
+		return "", false, nil
+	}
+	if tag == "" {
+		return "", true, nil
+	}
+	elems := strings.Split(tag, tagSep)
+	var opts []string
+	if len(elems) > 1 {
+		opts = elems[1:]
+	}
+	return elems[0], true, opts
+}
+
+func jsonFieldName(f reflect.StructField) (string, bool) {
+	s, ok, _ := parseTag(tagJSON, f)
+	return s, ok
+}
+
+// structToRule looks through the tags in a struct to (1) make sure there
+// are no duplicate tags since package json doesn't do this and it results
+// in hard-to-debug behaviors, and (2) it finds an "unpack" tag which indicates
+// which field of the struct to use as the key and what value to match for
+// unmarshling into this concrete type.  If there is not exactly one unpack
+// tag, then an error is returned.
+// unpack="key,skip" unpack="key", unpack="", or unpack="skip"
```

Comment:
I think the syntax ought to be `unpack=",skip"` just as `encoding/json` uses `json=",omitempty"`.

Reply:
I made it but didn't update the comment.  Done.